incubator/crate/values.yaml

```diff
@@ -0,0 +1,53 @@
+# The Chart for a crateDB deployment with working defaults consists of:
+#   1) A Service (exposing the ports that crate uses)
+#   2) A StatefulSet (initialising the cluster)
+#   3) Resources definition
+app: crate
+
+image:
+  name: crate
+  tag: 4.0.3
+  pullPolicy: IfNotPresent
+
+# 1) The Service connecting 3 ports.
+service:
+  name: crate
+  type: ClusterIP
+
+  adminName: crate-web
+  adminPort: 4200
+  clusterName: cluster
+  clusterPort: 4300
+  # https://crate.io/docs/crate/reference/en/latest/interfaces/postgres.html
+  postgresName: postgres
+  postgresPort: 5432
+
+# 2) The StatefulSet forming the cluster. Read more:
+# https://crate.io/docs/crate/guide/en/latest/scaling/multi-node-setup.html#id10
+# https://crate.io/docs/crate/reference/en/latest/config/environment.html#use-max-50-of-available-ram
+crate:
+  clusterName: my-cluster
+  crateHeapSize: 1g
+  recoverAfterNodes: 1
+  replicas: 1  # If you want more replicas, checkout the links above.
+  recoverAfterTime: 5m
+
+http:
+  cors:
+    enabled: false
+    allowOrigin: "*"
+
+# 3) Resources and Volumes.
+resources:
+  requests:
+    memory: 1Gi
+  limits:
+    memory: 2Gi
```

Comment:
What about also adding CPU request/limit?

Reply:
Hm, what about actually doing something with the memory request/limit while we're at it...!

Reply:
Resource limits are rendered into the template [here](https://github.com/helm/charts/pull/16318/files#diff-cf4b8dda181f1a7229191f17ca16be93R58) if that is what you mean?

Reply:
Other than that, I think `requests` and `limits` could (and should) have the same value, alternatively only provide `limits`.
The memory limit should be at least 2 times and up to 3 times higher than the allocated heap size for CrateDB. https://crate.io/docs/crate/guide/en/latest/performance/memory.html

I don't know whether it would somehow be possible to configure the required resource memory limit based on the CrateDB heap size parameter (or vice versa).

Reply:
Ah, I just noticed that there is a comment already regarding the heap configuration [here](https://github.com/helm/charts/pull/16318/files#diff-2708716e3dd4d331124ecac79a27ba6dR20).

Reply:
Ah - so it is there, bad grep, thanks.

Yes, we can do that. How about having configurable `crateHeapSize` (as it is) and `limits.memory`, but inferring the `requests.memory` as 2x heap size? And the _default_ `limits.memory` at 2x as well, or 3x?